auth/auth.go

```diff
@@ -115,7 +117,13 @@ func (n *NonceHandler) Nonce(
 		n.nonceCache = make(map[string]uint64)
 		//clear the cache for each block
 	}
-	seq := loomchain.NewSequence(nonceKey(origin)).Next(state)
+	var seq uint64
+	if state.FeatureEnabled(loomchain.IncrementNonceFailedTxFeature, false) && !isCheckTx {
+		// Only persist nonce for DeliverTx
```

Comment:
`Only persist nonce for DeliverTx` -> `Unconditionally increment the nonce in DeliverTx, regardless of whether the tx succeeds`

Reply:
Done